text/0182-otlp-remote-parent.md

```diff
@@ -0,0 +1,124 @@
+# Export SpanContext.IsRemote in OTLP
+
+Update OTLP to indicate whether a span's parent is remote.
+
+## Motivation
+
+It is sometimes useful to post-process or visualise only entry-point spans: spans which either have no parent (trace roots), or which have a remote parent.
+For example, the Elastic APM solution highlights entry-point spans (Elastic APM refers to these as "transactions") and surfaces these as top-level operations
+in its user interface.
+
+The goal is to identify the spans which represent a request that is entering a service, or originating within a service, without having to first assemble the
+complete distributed trace as a DAG (Directed Acyclic Graph). It is trivially possible to identify trace roots, but it is not possible to identify spans with
+remote parents.
+
+Here is a contrived example distributed trace, with a border added to the entry-point spans:
+
+```mermaid
+graph TD
+    subgraph comments_service
+    POST_comments(POST /comment)
+    POST_comments --> comments_send(comments send)
+    end
+
+    subgraph auth_service
+    POST_comments --> POST_auth(POST /auth)
+    POST_auth --> LDAP
+    end
+
+    subgraph user_details_service
+    POST_comments --> GET_user_details(GET /user_details)
+    GET_user_details --> SELECT_users(SELECT FROM users)
+    end
+
+    subgraph comments_inserter
+    comments_send --> comments_receive(comments receive)
+    comments_receive --> comments_process(comments process)
+    comments_process --> INSERT_comments(INSERT INTO comments)
+    end
+
+    style POST_comments stroke-width:4
+    style POST_auth stroke-width:4
+    style GET_user_details stroke-width:4
+    style comments_receive stroke-width:4
+```
+
+## Explanation
+
+The OTLP encoding for spans has a boolean `parent_span_is_remote` field for identifying whether a span's parent is remote or not.
+All OpenTelemetry SDKs populate this field, and backends may use it to identify a span as being an entry-point span.
+A span can be considered an entry-point span if it has no parent (`parent_span_id` is empty), or if `parent_span_is_remote` is true.
+
+## Internal details
+
+The first part would be to update the trace protobuf, adding a `boolean parent_span_is_remote` field to the
+[`Span` message](https://github.com/open-telemetry/opentelemetry-proto/blob/b43e9b18b76abf3ee040164b55b9c355217151f3/opentelemetry/proto/trace/v1/trace.proto#L84).
+
+[`SpanContext.IsRemote`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#isremote) identifies whether span context has been propagated from a remote parent.
```

Comment:
Are we implicitly assuming that every time we read context from context propagators, the `SpanContext`should have `IsRemote=true`?

If yes, Do we need to call this out in this proposal?

Reply:
This is not "implicitly assumed", it is already part of the spec. https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#isremote

Note that this OTEP does not propose any new concepts, it simply makes an existing concept exported via OTLP.

Reply:
One thing that came out from the discussions above is that in addition will need to add a new `hasRemoteParent` API (https://github.com/open-telemetry/oteps/pull/182#discussion_r970918867), so will update this OTEP to make it clear

Reply:
Thanks for referencing that, apparently I had promised but forgotten to submit a suggestion (it will be only a small clarification though)